File: bug19_appended_wrong_list/main.java
import java.nio.ByteBuffer;
import java.util.Base64;
import java.util.ArrayList;

public class main {
    public static void main(String[] args) {
        ArrayList<Long> evens = new ArrayList<>();
        ArrayList<Long> odds = new ArrayList<>();
        
        for (long n = 1; n <= 20; n++) {
            if (n % 2 == 0) {
                evens.add(n);
            } else {
                evens.add(n);
            }
        }
        
        if (odds.isEmpty()) {
            throw new IllegalArgumentException("ValueError: odds list is empty: all values were appended to evens by mistake");
        }
        
        long acc = 0;
        int min_len = Math.min(evens.size(), odds.size());
        for (int i = 0; i < min_len; i++) {
            long e = evens.get(i);
            long o = odds.get(i);
            acc ^= (e * o) << (i % 8);
            acc *= 0x9E3779B97F4A7C15L;
            acc = (acc << 11) | (acc >>> 53);
        }
        
        ByteBuffer buffer = ByteBuffer.allocate(8);
        buffer.putLong(acc);
        System.out.println(Base64.getEncoder().encodeToString(buffer.array()));
    }
}

File: bug19_appended_wrong_list/main.js
function run() {
    let evens = [];
    let odds = [];
    
    for (let n = 1n; n <= 20n; n++) {
        if (n % 2n === 0n) {
            evens.push(n);
        } else {
            evens.push(n);
        }
    }
    
    if (odds.length === 0) {
        throw new Error("ValueError: odds list is empty: all values were appended to evens by mistake");
    }
    
    let acc = 0n;
    let min_len = Math.min(evens.length, odds.length);
    for (let i = 0; i < min_len; i++) {
        let e = evens[i];
        let o = odds[i];
        let shiftLeft = BigInt(i % 8);
        acc ^= (e * o) << shiftLeft;
        acc = BigInt.asUintN(64, acc * 0x9E3779B97F4A7C15n);
        acc = BigInt.asUintN(64, (acc << 11n) | (acc >> 53n));
    }
    
    let buf = Buffer.alloc(8);
    buf.writeBigUInt64BE(acc, 0);
    console.log(buf.toString('base64'));
}
run();
